### PR TITLE
Collect network connectivity data on box for monitoring

### PR DESCRIPTION
**Pull request by jimyang2008**
https://github.com/firewalla/firecommit/issues/163

### CODE REVIEW

File: sensor/NetworkMonitorSensor.js [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+  async run() {
+    
+    log.info("run NetworkMonitorSensor ...");
+
+    /*
+     * apply policy upon policy change or startup
+     */
+    extensionManager.registerExtension(POLICY_KEYNAME, this, {
+      applyPolicy: this.applyPolicy,
+      start: this.start,
```
Comment (MelvinTo) [inline sensor/NetworkMonitorSensor.js:103]: start/stop is not required.

File: sensor/NetworkMonitorSensor.js [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+    }
+    return;
+  }
+
+  async applyPolicy(host, ip, policy) {
+    log.info(`Apply network monitor policy with host(${host}), ip(${ip}), policy(${policy})`);
+    try {
+        if (ip === '0.0.0.0') {
+            if (policy.state === true) {
+                if (fc.isFeatureOn(FEATURE_NETWORK_MONITOR, false)) {//compatibility: new firewlla, old app
```
Comment (MelvinTo) [inline sensor/NetworkMonitorSensor.js:294]: no need this backward compatible code any more, it was only used for super old features

File: net2/SysManager.js [hunk 3]
```diff
@@ -73,6 +73,7 @@ const f = require('../net2/Firewalla.js');
 const i18n = require('../util/i18n.js');
 
 const dns = require('dns');
+const { reduce } = require('bluebird');
```
Comment (MelvinTo) [inline net2/SysManager.js:76]: no need to import special reduce function, nodejs has native reduce

File: sensor/NetworkMonitorSensor.js (hunk 1 above)
Comment (jimyang2008) [reply sensor/NetworkMonitorSensor.js:136]: Removed as suggested

File: sensor/NetworkMonitorSensor.js (hunk 2 above)
Comment (jimyang2008) [reply sensor/NetworkMonitorSensor.js]: Removed as suggested

File: net2/SysManager.js (hunk 3 above)
Comment (jimyang2008) [reply net2/SysManager.js]: Removed as suggested

Comment (jimyang2008) [reply net2/SysManager.js]: Removed as suggested

File: sensor/NetworkMonitorSensor.js (hunk 1 above)
Comment (jimyang2008) [reply sensor/NetworkMonitorSensor.js:136]: Removed as suggested

File: sensor/NetworkMonitorSensor.js (hunk 2 above)
Comment (jimyang2008) [reply sensor/NetworkMonitorSensor.js]: removed as suggested

**Pull request merged**